src/telliot_feeds/queries/diva_protocol.py

```diff
@@ -100,3 +107,40 @@ def value_type(self) -> DIVAReturnType:
         - `packed`: false
         """
         return DIVAReturnType(abi_type="(ufixed256x18,ufixed256x18)", packed=False)
+
+    @property
+    def query_data(self) -> bytes:
+        """Encode the query type and parameters to create the query data."""
+        if self.poolId is None or self.divaDiamond is None or self.chainId is None:
+            raise ValueError(
+                "Missing required parameters: "
+                + f"{str(self.poolId)}, {self.divaDiamond}, {self.chainId}"  # type: ignore
+            )
+        param_types = [p["type"] for p in self.abi]
+        encoded_params = encode_abi(param_types, [bytes(self.poolId), self.divaDiamond, self.chainId])
```

Comment:
I don't think you have to cast to ```bytes(self.poolId)``` if HexBytes type since its already bytes, or did you? maybe for tox typing?

Reply:
I had to cast as bytes because encoding hexbytes was turning out different, I believe.